cognite/client/_api_client.py

```diff
@@ -804,16 +805,15 @@ def _convert_resource_to_patch_object(
             patch_object["externalId"] = dumped_resource.pop("externalId")
 
         for key, value in dumped_resource.items():
-            if utils._auxiliary.to_snake_case(key) in update_attributes:
+            if utils._text.to_snake_case(key) in update_attributes:
                 patch_object["update"][key] = {"set": value}
         return patch_object
 
     @staticmethod
     def _status_ok(status_code: int) -> bool:
         return status_code in {200, 201, 202, 204}
 
-    @classmethod
-    def _raise_api_error(cls, res: Response, payload: Dict) -> NoReturn:
+    def _raise_api_error(self, res: Response, payload: Dict) -> NoReturn:
```

Comment:
This can remain a classmethod?

Reply:
Why? 🤔  It is not used anywhere outside of APIClient.

Reply:
The code becomes easier to reason about when you can know that a method doesn't require access to any instance variables/methods just by looking at the definition. So IMHO it's good practice to default to using static/class methods.

Reply:
I agree with most of that. But when a function like this is _only_ dependent on other `staticmethod`s, it feels like a design flaw with the class. What I mean is that I don't like using classes to group a set of functions instead of having the as simple standalone ones.